app/assets/javascripts/models/beta/story.js

```diff
@@ -141,24 +146,60 @@ export const toggleStory = (story) => {
   };
 };
 
-const stateFor = (story) => {
-  if (story.state === "unscheduled" && typeof story.estimate === "number") return "unstarted";
-  if (story.state === "unstarted" && isUnestimatedFeature(story)) return "unscheduled";
-  if (!isFeature(story) && story.state === "unscheduled") return "unstarted";
-  return story.state;
+const isUnstartedState = (story, newAttributes) => 
+  !haveEstimate(story) && haveEstimate(newAttributes)
+
+const isUnscheduledState = (story, newAttributes) => 
+  (
+    isFeature(story._editing) && (
+      (
+        !haveEstimate(story._editing) &&
+        !haveEstimate(newAttributes)
+      ) || (
+        !haveEstimate(newAttributes) &&
+        has(newAttributes, 'estimate')
+      ) || (
+        isUnscheduled(story._editing) && 
+        !has(newAttributes, 'state') &&
+        !haveEstimate(newAttributes)
+      ) || isUnscheduled(newAttributes)
+    )
+  )
+
+const stateFor = (story, newAttributes, newStory) => {
+  const { UNSTARTED, UNSCHEDULED } = status;
+
+  if (isUnstartedState(story._editing, newAttributes)) return UNSTARTED;
+  if (isUnscheduledState(story, newAttributes)) return UNSCHEDULED;
+  
+  return newStory.state;
 }
 
+const estimateFor = (story, newAttributes, newStory) => {
+  if (isNoEstimated(story, newAttributes) || isUnscheduledState(story, newAttributes)) return '';
+  if (isFeature(newAttributes) && !isUnscheduled(story._editing)) return 1;
+  
+  return newStory.estimate;
+}
+
+const isEstimable = story => isFeature(story)
```

Comment:
You could rewrite this function like this:
```js
const isEstimable = isFeature;
```

Reply:
Thanks!